### PR TITLE
[BISERVER-12968]-Self Links defined in a PRPT chart do not work on the

### PR DESCRIPTION
**Pull request by PashkovichAndrei**
server

### CODE REVIEW

**Comment by PashkovichAndrei** [comment]
@tmorgner, could you please do review?

**Comment by pamval** [comment]
@tmorgner, please review when you get the chance

**Comment by PashkovichAndrei** [comment]
@tmorgner, @pamval, @mbatchelor, could you please do review?

File: core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java [hunk 1]
```diff
@@ -588,6 +591,14 @@ private String extractContentLinkSpec() {
     }
   }
 
+  static String getReportPath( MasterReport report ) {
+    ResourceManager rm = report != null ? report.getResourceManager() : null;
+    BundleResourceManagerBackend backend = rm != null ? (BundleResourceManagerBackend) rm.getBackend() : null;
```
Comment (tmorgner) [inline core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java]: You cannot assume that in 100% of the cases. Some weird person may actually call this action/component with a self-constructed MasterReport instance that has not been loaded from a bundle. Use an "instanceof" check here.

You may find it safer to work directly with the resource-key itself. Have a look at the while-loop in "PentahoKettleTransFormFileProducer#computeFullFilename". Combined with a similar loop in AbstractKettleTransformationProducer#computeFullFilename" testing for java.io.File objects, this is the safest solution to the problem and handles all sort of weird side-cases (like loading a prpt-report contained in a zip or jar file).

**Comment by PashkovichAndrei** [comment]
@tmorgner, could you please do review?

**Comment by tmorgner** [comment]
Let me reapeat this outside of a hidden "code review" comment: 

Adding the "instanceof" check is a good example for a "bare minimum fix". Please adapt your code to locate the resource-key in the same way the KettleTransformationProducer does it. Copying that while-loop is not that hard!

For reference: This is the earlier comment:
You may find it safer to work directly with the resource-key itself. Have a look at the while-loop in "PentahoKettleTransFormFileProducer#computeFullFilename". Combined with a similar loop in AbstractKettleTransformationProducer#computeFullFilename" testing for java.io.File objects, this is the safest solution to the problem and handles all sort of weird side-cases (like loading a prpt-report contained in a zip or jar file).

**Comment by PashkovichAndrei** [comment]
@tmorgner, could you please do review?

File: core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java [hunk 2]
```diff
@@ -588,6 +592,24 @@ private String extractContentLinkSpec() {
     }
   }
 
+  static String getReportPath( MasterReport report ) {
```
Comment (tmorgner) [inline core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java]: Try "report.getResourceKey()" for a change. That will give you the resource-key the report was loaded with without having to do all that unwrapping. Remember: If it is awfully complex, then you might be doing it wrong. 

    ResourceKey key = report.getResourceKey();

can replace all your first three lines of code, and you dont have to blind-cast the key-variable into a ResourceKey any more. How cool is that?

**Comment by PashkovichAndrei** [comment]
@tmorgner, could you please do review?

**Comment by PashkovichAndrei** [comment]
@tmorgner could you please have a look?

**Comment by tmorgner** [comment]
That looks good now.

**Pull request merged**